src/CameraOnvif.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef CAMERA_ONVIF_CAMERAONVIF_HPP
+#define CAMERA_ONVIF_CAMERAONVIF_HPP
+
+#include <string>
+
+namespace camera_onvif {
+    /**
+     *
+     */
+    class CameraOnvif {
+        struct Private;
+        Private* m_private = nullptr;
+
+        std::string user;
+        std::string pass;
+        std::string uri;
+        std::string video_token;
+        std::string video_conf_token;
+
+        float brightness[2]; // [Min, Max]
+        float color_saturation[2]; // [Min, Max]
+        float contrast[2]; // [Min, Max]
+
+        void setCredentials();
+        void reportError();
+        void init();
+
+    public:
+        CameraOnvif(/* args */);
+        CameraOnvif(std::string user, std::string pass, std::string ip);
```

Comment:
Whenever you pass non scalars as parameters to methods and functions, prefer passing them as const refs (if they are input parameters) or pointers (if they are output parameters). This makes your code more efficient because you won't be copying everything around at each call. Won't make much of a difference here because a string is a lightweight object but it could have a major impact if you were passing around images, for example